### ISSUE TITLE
Properly fix a serious crasher

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#1161]
Recent changes work around a serious crash, given below.  A proper fix would deal with the root of the problem.
```python
Traceback (most recent call last):
  File "c:\leo.repo\leo-editor\leo\plugins\qt_events.py", line 93, in eventFilter
    g.es_exception()
  File "c:\leo.repo\leo-editor\leo\core\leoGlobals.py", line 6238, in es_exception
    g.es_print_error(line, color=color)
  File "c:\leo.repo\leo-editor\leo\core\leoGlobals.py", line 6216, in es_print_error
    g.es_print(*args, **keys)
  File "c:\leo.repo\leo-editor\leo\core\leoGlobals.py", line 6258, in es_print
    g.es(*args, **keys)
  File "c:\leo.repo\leo-editor\leo\core\leoGlobals.py", line 6164, in es
    log.put(s, color=color, tabName=tabName, nodeLink=d['nodeLink'])
  File "c:\leo.repo\leo-editor\leo\plugins\qt_frame.py", line 3585, in put
    w = self.logCtrl.widget # w is a QTextBrowser
AttributeError: 'todoQtUI' object has no attribute 'widget'
```

**Comment by edreamleo** [comment]
Clearly, the todo.py plugin is somehow responsible for the logCtrl ivar, but I don't see how.

**Comment by edreamleo** [comment]
Fixed at rev afd704e.  This is a _major_ improvement in the code.  The checkin log:
```
LeoQtLog class ensures that self.logCtrl is *always* a QTextEditWrapper.
This fixes selection bugs, removes confusions and adds defensive code.

- Major change: LeoQtLog.createTab sets self.logCtrl as in legacy non-dock code.
  and is more carefull about setting widget.leo_log_wrapper.
- Major change: LeoQtLog.selectHelper sets self.logCtrl in a new way.
- Added defensive code to onCurrentChanged, put & putnl.

All tests pass regardless of --no-dock.

g.es('whatever') always shows the Log pane within a raised Tabs dock.
```
